cmake/developer-options.cmake

```diff
@@ -0,0 +1,171 @@
+#
```

Comment:
simdjson-options was getting pretty big, thanks for splitting it up :)

Reply:
More to come in later PRs!